core/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,273 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.history
+
+import java.io.{FileNotFoundException, IOException, OutputStream}
+import java.util.UUID
+import java.util.concurrent.{Executors, ExecutorService, Future, TimeUnit}
+import java.util.zip.{ZipEntry, ZipOutputStream}
+
+import scala.collection.mutable
+import scala.xml.Node
+
+import com.google.common.io.ByteStreams
+import com.google.common.util.concurrent.{MoreExecutors, ThreadFactoryBuilder}
+import org.apache.hadoop.fs.{FileStatus, Path}
+import org.apache.hadoop.fs.permission.FsAction
+import org.apache.hadoop.hdfs.DistributedFileSystem
+import org.apache.hadoop.hdfs.protocol.HdfsConstants
+import org.apache.hadoop.security.AccessControlException
+
+import org.apache.spark.{SecurityManager, SparkConf, SparkException}
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.internal.Logging
+import org.apache.spark.scheduler._
+import org.apache.spark.scheduler.ReplayListenerBus._
+import org.apache.spark.ui.SparkUI
+import org.apache.spark.util.{Clock, SystemClock, ThreadUtils, Utils}
+
+/**
+ * A class that provides application history from event logs stored in the file system.
+ * This provider will render the history application UI of specifically requested
+ * applications only.  No scanning or caching of event logs from a filesystem is performed.
+ *
+ */
+
+private[history] class SimpleFsHistoryProvider(conf: SparkConf)
+  extends ApplicationHistoryProvider with Logging {
+
+  import SimpleFsHistoryProvider._
+
+  private val logDir = conf.getOption("spark.history.fs.logDirectory")
+    .getOrElse(DEFAULT_LOG_DIR)
+
+  private val HISTORY_UI_ACLS_ENABLE = conf.getBoolean("spark.history.ui.acls.enable", false)
+  private val HISTORY_UI_ADMIN_ACLS = conf.get("spark.history.ui.admin.acls", "")
+  private val HISTORY_UI_ADMIN_ACLS_GROUPS = conf.get("spark.history.ui.admin.acls.groups", "")
+  logInfo(s"History server ui acls " + (if (HISTORY_UI_ACLS_ENABLE) "enabled" else "disabled") +
+    "; users with admin permissions: " + HISTORY_UI_ADMIN_ACLS.toString +
+    "; groups with admin permissions" + HISTORY_UI_ADMIN_ACLS_GROUPS.toString)
+
+  private val hadoopConf = SparkHadoopUtil.get.newConfiguration(conf)
+  private val fs = new Path(logDir).getFileSystem(hadoopConf)
```

Comment:
This seems awkward - why not `private val fs = <org.apache.hadoop.fs.>FileSystem.get(hadoopConf)`?